### PR TITLE
Add filtering

### PR DESCRIPTION
**Pull request by a96lex**

### CODE REVIEW

**Review by zeritte — CHANGES_REQUESTED** [review]
there are some changes need to be done:
- removing the console.logs
- adding the other endpoint => Get voice parameters (https://v1.api.audio/voice/parameter) 
maybe smt like: `Aflr.Voice.parameters()`
- adding missing tests

but most of it is there and thanks for the PR :)

File: src/__tests__/Voice.test.ts [hunk 1]
```diff
@@ -32,7 +32,8 @@ describe("Voice operations", () => {
 
   test("It should list all of the scripts and find the created one", async () => {
     try {
-      const voices: any = await Voice.list();
+      const voices: any = await Voice.list({ industryExamples: "education" });
```
[the remark this answers is not in the document]
Comment (a96lex) [reply src/__tests__/Voice.test.ts:35]: it works

File: src/__tests__/Voice.test.ts [hunk 2]
```diff
@@ -32,7 +32,8 @@ describe("Voice operations", () => {
 
   test("It should list all of the scripts and find the created one", async () => {
```
[the remark this answers is not in the document]
Comment (a96lex) [reply src/__tests__/Voice.test.ts]: edited and added other test

File: src/__tests__/Voice.test.ts [hunk 3]
```diff
@@ -39,4 +39,14 @@ describe("Voice operations", () => {
       throw new Error("test failed");
     }
   });
+
+  test("It should list all of the voices that match the filtering parameters", async () => {
+    try {
+      const voices: any = await Voice.list({ tags: "fun", gender: "male", language: "english" });
+      expect(Array.isArray(voices.voices)).toBe(true);
```
Comment (zeritte) [inline src/__tests__/Voice.test.ts:46]: hey thanks for the update, it looks far better now! just one more thing, do you think we should at least compare the number of voices filtered/non-filtered?
the other test without filters can save that number to a variable, then at the end of this test with filtering enabled, we compare two and say `expect(voices.voices.length).toBeLessThan(allVoicesCount)`

**Review by zeritte — APPROVED** [review]
looks better now! I suggested another improvement if you want to spend some time on this

File: src/__tests__/Voice.test.ts (hunk 3 above)
Comment (a96lex) [reply src/__tests__/Voice.test.ts:49]: compared

Comment (a96lex) [reply src/__tests__/Voice.test.ts:49]: I added toBeLessOrEqual, just in case the filtering parameter is the same for all voices. Will never happen I think but it is an edge case covered

File: src/__tests__/Voice.test.ts [hunk 4]
```diff
@@ -30,10 +30,34 @@ describe("Voice operations", () => {
...
+    } catch (e) {
+      console.error(e);
+      throw new Error("test failed");
+    }
+  });
+
+  test("It should list all of the available filtering parameters", async () => {
+    try {
+      const parameters: any = await Voice.parameters();
+      expect(parameters !== null && typeof parameters === "object").toBe(true);
```
Comment (zeritte) [inline src/__tests__/Voice.test.ts:60]: you can remove the !== null I think if you have the object type.

and you can do better than this here in the test, you know the return type. you can say it must include everything in the IVoiceFilteringBody

File: src/Voice.ts [hunk 5]
```diff
@@ -1,30 +1,42 @@
...
 import { isInitializedError, isSubmoduleAlreadyInitializedError } from "./Errors";
 import { RequestBase } from "./RequestBase";
-import { IConfig } from "./types";
+import { IConfig, IVoiceFilteringBody } from "./types";
 
 export class VoiceClass {
   #initialized = false;
   #RequestClass!: RequestBase;
   #url = "";
+  #url_parameters = "";
```
Comment (zeritte) [inline src/Voice.ts:10]: the convention I was following was `resourcename_url` so it would be great if you could change this to `parameters_url`

**Review by zeritte — CHANGES_REQUESTED** [review]
we are getting there but still a few more flaws to be perfect :)

File: src/Voice.ts (hunk 5 above)
Comment (a96lex) [reply src/Voice.ts:10]: :+1:

**Review by zeritte — APPROVED** [review]
